### PR TITLE
Pan os add filter to rules commands

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5231

## Description
Added some filters and outputs for rule-commands.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 1]
```diff
@@ -5627,9 +5627,35 @@ script:
       name: device-group
     - description: The tag to filter the rules.
       name: tag
+      isArray: true
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:5630]: If this is true, then shouldn't the description be: A comma-separated list of tags by which to filter the rules.

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 2]
```diff
@@ -8998,6 +9055,31 @@ script:
...
+    - auto: PREDEFINED
+      description: The action of the PBF rules to retrieve. If not mentioned will retrieve all the PBF rules.
+      name: action
+      isArray: false
+      required: false
+      predefined:
+      - discard
+      - forward
+      - no-pbf
+    - description: The tags of the PBF rules to retrieve. If not mentioned will retrieve all the PBF rules.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:9075]:
```suggestion
    - description: A comma-separated list of tags of the PBF rules to retrieve. If not mentioned will retrieve all the PBF rules.
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 3]
```diff
@@ -8453,6 +8482,31 @@ script:
...
+    - auto: PREDEFINED
+      description: The type of the NAT rules to retrieve. If not mentioned will retrieve all the NAT rules.
+      name: nat_type
+      isArray: false
+      required: false
+      predefined:
+      - ipv4
+      - nat64
+      - nptv6
+    - description: The tags of the NAT rules to retrieve. If not mentioned will retrieve all the NAT rules.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:8502]:
```suggestion
    - description: A comma-separated list of tags of the NAT rules to retrieve. If not mentioned will retrieve all the NAT rules.
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 4]
```diff
@@ -5627,9 +5627,35 @@ script:
...
+      isArray: true
     - description: Serial number of the firewall on which to run the command. Use only for a Panorama instance.
       name: target
       required: false
+    - description: The name of the rule to retrieve. If not mentioned, will retrieve all the rules.
+      isArray: false
+      name: rulename
+      required: false
+    - auto: PREDEFINED
+      description: Whether to retrieve the disabled rules or not. If not mentioned will retrieve all the rules.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:5639]:
```suggestion
      description: Whether to retrieve the disabled rules or not. If not mentioned, will retrieve all the rules.
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 5]
```diff
@@ -5627,9 +5627,35 @@ script:
...
+      name: disabled
+      isArray: false
+      required: false
+      predefined:
+      - yes
+      - no
+    - auto: PREDEFINED
+      description: The action of the rules to retrieve. If not mentioned will retrieve all the rules.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:5647]:
```suggestion
      description: The action of the rules to retrieve. If not mentioned, will retrieve all the rules.
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 6]
```diff
@@ -5627,9 +5627,35 @@ script:
...
+      name: action
+      isArray: false
+      required: false
+      predefined:
+      - allow
+      - deny
+      - drop
+    - description: Free query to retrieve rules. If not mentioned will retrieve all the rules.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:5655]:
```suggestion
    - description: Free query to retrieve rules. If not mentioned, will retrieve all the rules.
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 7]
```diff
@@ -8453,6 +8482,31 @@ script:
       predefined:
       - 'true'
       - 'false'
+    - auto: PREDEFINED
+      description: Whether to retrieve the disabled rules or not. If not mentioned will retrieve all the NAT rules.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:8486]:
```suggestion
      description: Whether to retrieve the disabled rules. If not mentioned, will retrieve all the NAT rules.
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 8]
```diff
@@ -8453,6 +8482,31 @@ script:
...
+      name: disabled
+      isArray: false
+      required: false
+      predefined:
+      - yes
+      - no
+    - auto: PREDEFINED
+      description: The type of the NAT rules to retrieve. If not mentioned will retrieve all the NAT rules.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:8494]:
```suggestion
      description: The type of the NAT rules to retrieve. If not mentioned, will retrieve all the NAT rules.
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 9]
```diff
@@ -8998,6 +9055,31 @@ script:
       - 'false'
       name: show_uncommitted
       required: false
+    - auto: PREDEFINED
+      description: Whether to retrieve the disabled rules or not. If not mentioned will retrieve all the PBF rules.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:9059]:
```suggestion
      description: Whether to retrieve the disabled rules. If not mentioned, will retrieve all the PBF rules.
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 10]
```diff
@@ -8998,6 +9055,31 @@ script:
...
+      name: disabled
+      isArray: false
+      required: false
+      predefined:
+      - yes
+      - no
+    - auto: PREDEFINED
+      description: The action of the PBF rules to retrieve. If not mentioned will retrieve all the PBF rules.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:9067]:
```suggestion
      description: The action of the PBF rules to retrieve. If not mentioned, will retrieve all the PBF rules.
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 11]
```diff
@@ -8998,6 +9055,31 @@ script:
...
+      required: false
+      predefined:
+      - discard
+      - forward
+      - no-pbf
+    - description: The tags of the PBF rules to retrieve. If not mentioned will retrieve all the PBF rules.
+      name: tags
+      isArray: true
+      required: false
+    - description: Free query to retrieve PBF rule. If not mentioned will retrieve all the PBF rules.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:9079]:
```suggestion
    - description: Free query to retrieve PBF rule. If not mentioned, will retrieve all the PBF rules.
```

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 12]
```diff
@@ -3029,6 +3029,10 @@ Returns a list of predefined Security Rules.
 | device-group | The device group for which to return addresses (Panorama instances). | Optional | 
 | tag | Tag for which to filter the rules. | Optional | 
 | target | Serial number of the firewall on which to run the command. Use only for a Panorama instance | Optional |
+| rulename | The name of the rule to retrieve. If not mentioned, will retrieve all the rules. | Optional | 
+| disabled | Whether to retrieve the disabled rules or not. If not mentioned will retrieve all the rules. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/README.md:3033]:
```suggestion
| disabled | Whether to retrieve the disabled rules. If not mentioned, will retrieve all the rules. | Optional | 
```

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 13]
```diff
@@ -3029,6 +3029,10 @@ Returns a list of predefined Security Rules.
 | device-group | The device group for which to return addresses (Panorama instances). | Optional | 
 | tag | Tag for which to filter the rules. | Optional | 
 | target | Serial number of the firewall on which to run the command. Use only for a Panorama instance | Optional |
+| rulename | The name of the rule to retrieve. If not mentioned, will retrieve all the rules. | Optional | 
+| disabled | Whether to retrieve the disabled rules or not. If not mentioned will retrieve all the rules. | Optional | 
+| action | The action of the rules to retrieve. If not mentioned will retrieve all the rules. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/README.md:3034]:
```suggestion
| action | The action of the rules to retrieve. If not mentioned, will retrieve all the rules. | Optional | 
```

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 14]
```diff
@@ -3029,6 +3029,10 @@ Returns a list of predefined Security Rules.
 | device-group | The device group for which to return addresses (Panorama instances). | Optional | 
 | tag | Tag for which to filter the rules. | Optional | 
 | target | Serial number of the firewall on which to run the command. Use only for a Panorama instance | Optional |
+| rulename | The name of the rule to retrieve. If not mentioned, will retrieve all the rules. | Optional | 
+| disabled | Whether to retrieve the disabled rules or not. If not mentioned will retrieve all the rules. | Optional | 
+| action | The action of the rules to retrieve. If not mentioned will retrieve all the rules. | Optional | 
+| query | Free query to retrieve rules. If not mentioned will retrieve all the rules. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/README.md:3035]:
```suggestion
| query | Free query to retrieve rules. If not mentioned, will retrieve all the rules. | Optional |
```

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 15]
```diff
@@ -7678,6 +7686,10 @@ Returns a list of NAT rules of either a Panorama/firewall instance.
 | device-group | The device group in which the NAT rules are part of. | Optional | 
 | pre_post | The pre-rule or post-rule (Panorama instances only). Possible values are: pre-rulebase, post-rulebase. | Optional | 
 | show_uncommitted | Whether to show the un-committed rules or not. can be true or false. Default is false. | Optional | 
+| disabled | Whether to retrieve the disabled rules or not. If not mentioned will retrieve all the NAT rules. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/README.md:7689]:
```suggestion
| disabled | Whether to retrieve the disabled rules. If not mentioned, will retrieve all the NAT rules. | Optional | 
```

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 16]
```diff
@@ -7678,6 +7686,10 @@ Returns a list of NAT rules of either a Panorama/firewall instance.
 | device-group | The device group in which the NAT rules are part of. | Optional | 
 | pre_post | The pre-rule or post-rule (Panorama instances only). Possible values are: pre-rulebase, post-rulebase. | Optional | 
 | show_uncommitted | Whether to show the un-committed rules or not. can be true or false. Default is false. | Optional | 
+| disabled | Whether to retrieve the disabled rules or not. If not mentioned will retrieve all the NAT rules. | Optional | 
+| nat_type | The type of the NAT rules to retrieve. If not mentioned will retrieve all the NAT rules. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/README.md:7690]:
```suggestion
| nat_type | The type of the NAT rules to retrieve. If not mentioned, will retrieve all the NAT rules. | Optional | 
```

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 17]
```diff
@@ -7678,6 +7686,10 @@ Returns a list of NAT rules of either a Panorama/firewall instance.
 | device-group | The device group in which the NAT rules are part of. | Optional | 
 | pre_post | The pre-rule or post-rule (Panorama instances only). Possible values are: pre-rulebase, post-rulebase. | Optional | 
 | show_uncommitted | Whether to show the un-committed rules or not. can be true or false. Default is false. | Optional | 
+| disabled | Whether to retrieve the disabled rules or not. If not mentioned will retrieve all the NAT rules. | Optional | 
+| nat_type | The type of the NAT rules to retrieve. If not mentioned will retrieve all the NAT rules. | Optional | 
+| tags | The tags of the NAT rules to retrieve. If not mentioned will retrieve all the NAT rules. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/README.md:7691]:
```suggestion
| tags | The tags of the NAT rules to retrieve. If not mentioned, will retrieve all the NAT rules. | Optional | 
```

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 18]
```diff
@@ -7678,6 +7686,10 @@ Returns a list of NAT rules of either a Panorama/firewall instance.
 | device-group | The device group in which the NAT rules are part of. | Optional | 
 | pre_post | The pre-rule or post-rule (Panorama instances only). Possible values are: pre-rulebase, post-rulebase. | Optional | 
 | show_uncommitted | Whether to show the un-committed rules or not. can be true or false. Default is false. | Optional | 
+| disabled | Whether to retrieve the disabled rules or not. If not mentioned will retrieve all the NAT rules. | Optional | 
+| nat_type | The type of the NAT rules to retrieve. If not mentioned will retrieve all the NAT rules. | Optional | 
+| tags | The tags of the NAT rules to retrieve. If not mentioned will retrieve all the NAT rules. | Optional | 
+| query | Free query to retrieve NAT rule. If not mentioned will retrieve all the NAT rules. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/README.md:7692]:
```suggestion
| query | Free query to retrieve NAT rule. If not mentioned, will retrieve all the NAT rules. | Optional | 
```

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 19]
```diff
@@ -8279,6 +8294,10 @@ Returns a list of pbf-rules of either a Panorama/firewall instance.
 | device-group | The device-group that the pbf-rules are part of. | Optional | 
 | pre_post | The pre-rule or post-rule (Panorama instances only). Possible values are: pre-rulebase, post-rulebase. | Optional | 
 | show_uncommitted | Whether to show the un-committed rules or not. can be true or false. Default is false. | Optional | 
+| disabled | Whether to retrieve the disabled rules or not. If not mentioned will retrieve all the PBF rules. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/README.md:8297]:
```suggestion
| disabled | Whether to retrieve the disabled rules. If not mentioned, will retrieve all the PBF rules. | Optional | 
```

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 20]
```diff
@@ -8279,6 +8294,10 @@ Returns a list of pbf-rules of either a Panorama/firewall instance.
 | device-group | The device-group that the pbf-rules are part of. | Optional | 
 | pre_post | The pre-rule or post-rule (Panorama instances only). Possible values are: pre-rulebase, post-rulebase. | Optional | 
 | show_uncommitted | Whether to show the un-committed rules or not. can be true or false. Default is false. | Optional | 
+| disabled | Whether to retrieve the disabled rules or not. If not mentioned will retrieve all the PBF rules. | Optional | 
+| action | The action of the PBF rules to retrieve. If not mentioned will retrieve all the PBF rules. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/README.md:8298]:
```suggestion
| action | The action of the PBF rules to retrieve. If not mentioned, will retrieve all the PBF rules. | Optional | 
```

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 21]
```diff
@@ -8279,6 +8294,10 @@ Returns a list of pbf-rules of either a Panorama/firewall instance.
 | device-group | The device-group that the pbf-rules are part of. | Optional | 
 | pre_post | The pre-rule or post-rule (Panorama instances only). Possible values are: pre-rulebase, post-rulebase. | Optional | 
 | show_uncommitted | Whether to show the un-committed rules or not. can be true or false. Default is false. | Optional | 
+| disabled | Whether to retrieve the disabled rules or not. If not mentioned will retrieve all the PBF rules. | Optional | 
+| action | The action of the PBF rules to retrieve. If not mentioned will retrieve all the PBF rules. | Optional | 
+| tags | The tags of the PBF rules to retrieve. If not mentioned will retrieve all the PBF rules. | Optional | 
+| query | Free query to retrieve PBF rule. If not mentioned will retrieve all the PBF rules. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/README.md:8300]:
```suggestion
| query | Free query to retrieve PBF rule. If not mentioned, will retrieve all the PBF rules. | Optional | 
```

File: Packs/PAN-OS/ReleaseNotes/1_16_7.md [hunk 22]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Palo Alto Networks PAN-OS
+- Added filters for ***pan-os-list-rules*** ***pan-os-list-nat-rules*** ***pan-os-list-pbf-rules*** commands.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/ReleaseNotes/1_16_7.md:4]:
```suggestion
- Added filters for the following commands:
   - ***pan-os-list-rules*** 
   - ***pan-os-list-nat-rules***
   - ***pan-os-list-pbf-rules***
```

**Comment by ShirleyDenkberg** [comment]
@GuyAfik Doc review completed.

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 23]
```diff
@@ -3473,19 +3499,29 @@ def panorama_list_rules_command(args: dict):
...
+    )
+    name = args.get('rulename')
+    query = args.get('query')
     target = args.get('target')
-    rules = panorama_list_rules(xpath, tag)
+
+    rules = panorama_list_rules(xpath, name, filters, query)
     pretty_rules = prettify_rules(rules, target)
 
+    demisto.debug('This is the pretty rules')
```
Comment (gal-forer) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:3514]:
```suggestion
    demisto.debug('Those are the pretty rules')
```

**Review by gal-forer — APPROVED** [review]

**Review by gal-forer — APPROVED** [review]

**Pull request merged**